genmc-sys/src/lib.rs

```diff
@@ -163,6 +188,19 @@ mod ffi {
         is_coherence_order_maximal_write: bool,
     }
 
+    #[must_use]
+    #[derive(Debug)]
+    struct ReadModifyWriteResult {
+        /// If there was an error, it will be stored in `error`, otherwise it is `None`.
+        error: UniquePtr<CxxString>,
+        /// The value that was read by the RMW operation as the left operand.
+        old_value: GenmcScalar,
+        /// The value that was produced by the RMW operation.
+        new_value: GenmcScalar,
+        /// `true` if the write should also be reflected in Miri's memory representation.
+        is_coherence_order_maximal_write: bool,
+    }
+
     /**** Types shared between Miri/Rust and GenMC/C++ through cxx_bridge: ****/
```

Comment:
```suggestion
    /**** These are GenMC types that we have to copy-paste here since cxx does not support
      "importing" externally defined C++ types. ****/
```